refactoringminer/src/main/java/gr/uom/java/xmi/decomposition/Visitor.java

```diff
@@ -44,329 +43,454 @@
+        if (value == null) {
+            addNullLiteral(source);
+        } else if (value instanceof String) {
+            addStringLiteral(source);
+        } else if (value instanceof Number) {
+            addNumberLiteral(source);
+        } else if (value instanceof Boolean) {
+            addBooleanLiteral(source);
+        } else {
+            // Characters not processed
+            assert value instanceof Character;
+        }
+    }
+
+    /**
+     * @param element TypeKeyword or PsiJavaCodeReferenceElement
+     */
+    private void processTypeIdentifier(@NotNull PsiElement element) {
+        int arrayDimensions = TypeUtils.arrayDimensions(element);
+        String typeStr = Formatter.format(element) + Strings.repeat("[]", arrayDimensions);
+        addType(typeStr);
+    }
+
+    private boolean processDeclaration(@NotNull PsiDeclarationStatement declaration) {
+        for (PsiElement declaredElement : declaration.getDeclaredElements()) {
+            if (declaredElement instanceof PsiVariable) {
+                VariableDeclaration variableDeclaration =
+                    new VariableDeclaration(file, filePath, (PsiVariable) declaredElement);
+                addVariableDeclaration(variableDeclaration);
+            } else if (declaredElement instanceof PsiClass) {
+                // Local classes are not yet supported
+                return false;
+            } else {
+                throw new AssertionError("Unqualified declared element");
+            }
+        }
+        return true;
+    }
+
+    private void processReference(@NotNull PsiReferenceExpression element) {
+        if (element.getParent() instanceof PsiMethodCallExpression
+            || element instanceof PsiMethodReferenceExpression) {
+            return;
+        }
+        List<String> reference = getReferenceTokens(element);
+        if (reference != null) {
+            int firstNotTypeIndex = 0;
+            for (int i = 0; i < reference.size(); i++) {
+                if (isClassName(reference.get(i))) {
+                    firstNotTypeIndex = i + 1;
+                }
+            }
+            if (firstNotTypeIndex != 0) {
+                String type = String.join(".", reference.subList(0, firstNotTypeIndex));
+                addType(type);
+            }
+            if (firstNotTypeIndex != reference.size()) {
+                String variable = String.join(".", reference);
+                addVariable(variable);
+            }
+        }
+    }
+
+    private boolean processNewExpression(@NotNull PsiNewExpression newExpression) {
+        ObjectCreation creation = new ObjectCreation(file, filePath, newExpression);
+        String source = Formatter.format(newExpression);
+        addCreation(creation, source);
+        if (newExpression.isArrayCreation()) {
+            PsiArrayInitializerExpression initializer = newExpression.getArrayInitializer();
+            return initializer == null || initializer.getInitializers().length <= 10;
```

Comment:
`initializer.getInitializers().length <= 10`
it is not clear why we check it

Reply:
Copied from the original instrument. Maybe because processing long lists of such elements is not important, but it takes time. Removing this condition does not affect the precision and recall